onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2289,14 +2289,33 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
           // Deserialize profile
           shape_ranges = DeserializeProfileV2(profile_file);
           LOGS_DEFAULT(VERBOSE) << "[TensorRT EP] DeSerialized " + profile_cache_path;
+
```

Comment:
i don't think this will work.  there's an assumption in the code that encrypted engine has a different name from unencrypted engine. if engine_file exists, that means there's an unencrypted engine. unencrypted engine shouldn't exist on disk.

Reply:
Hi George.
The encryption and model storage happens in between lines 2503-2515.
For both encrypted/plain engine, the engine_cache_path is the same.
So, the assumption is that there's only one engine (encrypted or plain) file for each model.

Weird, I don't see a "trt_engine_decryption_enable" and "trt_engine_decryption_lib_path" in https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#configurations anymore. 
But it still states: Warning: Please clean up any old engine and profile cache files.

I don't see, where the "assumption in the code that encrypted engine has a different name from unencrypted engine" is. 

"engine_decryption_enable" switches between loading a plain or encrypted engine.

Reply:
I didn't have the full context when the encryption/decryption workflows were added. But looking at the thread of discussion in #15519, I think the assumption "encrypted engine has a different name from unencrypted engine" is inside the encryption/decryption library where encryption function might create a real encrypted engine with different name even though the encryption function consumes "engine_cache_path". And decryption function knows which engine file to decrypt.

Agree with you guys that current TRT EP has an issue that "there was no code execution path possible so that an encrypted TRT engine cache could be created when flags engine_cache_enable and engine_decryption_enable are true"

Your code seems might fix this issue but there are some parts need to be changed as well:
- In the same block, the "else if" [path](https://github.com/AndreyOrb/onnxruntime/blob/main/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L2337)  seems not necessary now.
- Same logic should be applied [here](https://github.com/AndreyOrb/onnxruntime/blob/main/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L2055) as well.